### PR TITLE
Implement two-package structure

### LINKED ISSUE #73: CONTRIBUTING guidelines document
For discussion of policies/concepts to fill our CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by j08lue**
This pull request implements the changes we proposed in https://github.com/DHI/react-components/issues/73#issuecomment-788216167 a couple of weeks back. It aims at disrupting the structure, CI, and practice of the existing repo as little as possible.

Our hope is that DHI web devs with this can share components in one repo, learn from each other, build on each other's work, and be one step closer to an integrated DHI web dev community. There are certainly more steps to take, but maybe this can be a beginning.

## Changes and additions

- [x] Move existing components to a separate package (referred to as **core**)
- [x] Add a **lab** package for new components that qualify against the check list
- [x] Add the proposed checklist for the in/out decision to a pull request template that will show on all PRs
- [x] Add CONTRIBUTING documents for core and labs packages Closes #73
- [x] Add a CI pipeline and semantic versioning for the lab package
- [x] Add a storybook and stories for the lab package 

## Additional considerations

1. Currently, the new **core** package contains all the components that existed in this repo until now. In principle, these have not gone through the evaluation of style, quality, and whether they are reused sufficiently (which also requires generality and exposed props). We could also move the existing components to a third package and have a **core** package only for components that went through the lab lifecycle.
2. We need to discuss the review process for new submissions. An idea would be to require that any component submitted needs to be reviewed by at least two maintainers. This could even be enforced through a [branch protection rule](https://docs.github.com/en/github/administering-a-repository/managing-a-branch-protection-rule)

## Discussion questions

1. What do you think of this, @goneale, @JeanLoureiro, @nfour, and @FranzThomsen1089 
2. Do we need to cover more with this PR or take those in the next iteration?

## WIP

We are already adding a few more components to the **lab** package and have an open PR fixing the merge conflicts. Should very soon be all good to go.

- [x] Fix merge conflicts (happening in https://github.com/DHI-GRAS/react-components-ts/pull/5)
- [x] Add another lab component (courtesy of @rosudhi)

### CODE REVIEW

**Comment by j08lue** [comment]
I think I messed up the merge conflict resolution, dropping some of the changes you had made since we forked off a couple of weeks ago. That git-fu... I'll of course make sure this is all double-checked before removing the "WIP" on this PR.

**Comment by j08lue** [comment]
Almost all merge conflicts are from work below `/src`, so it is easy to make sure that the changes get preserved in the new location of that code by just taking a fresh copy. Should all be done correctly in https://github.com/DHI-GRAS/react-components-ts/pull/5. When we merge that, there should be no more conflicts.

File: README.md [hunk 1]
```diff
@@ -1,56 +1,35 @@
...
 
-React-Components are demonstrated [here](https://domainservices.dhigroup.com/) and are available as npm package from Github packages.
 
-## Getting started with development
+## Packages and their scope
 
-The recommended workflow is to run TSDX in one terminal:
+We have two packages, similar to what you might know from [Material-UI](https://material-ui.com/components/about-the-lab/):
+1. [Lab package](packages/react-components-lab) for rapid uptake and incubation of new components to prove their use and quality. [![Storybook](https://raw.githubusercontent.com/storybookjs/brand/master/badge/badge-storybook.svg)](https://storybooklab.z16.web.core.windows.net/)
+1. [Core package](packages/react-components) for mature and widely used components. This package also contains larger components from SPOPX. [![Storybook](https://raw.githubusercontent.com/storybookjs/brand/master/badge/badge-storybook.svg)](https://domainservices.dhigroup.com/)
```
Comment (FranzThomsen1089) [inline README.md:15]: Its true that they were built in SPOPX, but they are used in lots of other places as well. I think you should just leave it out

File: packages/react-components-lab/.releaserc.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{
+    "branches": [
+        "add-lab"
+    ],
+    "repositoryUrl": "https://github.com/DHI-GRAS/react-components-ts",
```
Comment (FranzThomsen1089) [inline packages/react-components-lab/.releaserc.json:5]: ?

File: packages/react-components-lab/docs/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+## [0.2.11](https://github.com/DHI-GRAS/react-components-ts/compare/v0.2.10...v0.2.11) (2021-03-12)
```
Comment (FranzThomsen1089) [inline packages/react-components-lab/docs/CHANGELOG.md:1]: Seems to be a lot of references to the other repo

Comment (goneale) [reply packages/react-components-lab/docs/CHANGELOG.md:1]: Yes it's all `react-components` now.

**Comment by goneale** [comment]
Yes, looks and sounds really good and appreciate the contribution. The idea of core and labs is a good approach.

If there are no merge conflicts, and you have ensured you have incorporated the latest master branch onto this repo as @JeanLoureiro recently included some new PRs to master, then this sounds good

File: packages/react-components-lab/.releaserc.json (hunk 2 above)
Comment (j08lue) [reply packages/react-components-lab/.releaserc.json:5]: I meant to mention that. We will fix the URLs in a quick follow-up PR to this one.

File: packages/react-components-lab/docs/CHANGELOG.md (hunk 3 above)
Comment (j08lue) [reply packages/react-components-lab/docs/CHANGELOG.md:1]: Correct. We will weed out those in a quick follow-up PR. Raised here: #87

File: README.md (hunk 1 above)
Comment (j08lue) [reply README.md]: I removed the reference.

**Comment by j08lue** [comment]
Thanks for the reviews @goneale and @FranzThomsen1089.

The PR already incorporates the latest commits on `master`, so you can merge it at your earliest convenience and we will do the remaining (repo URL) clean-up very soon after.

Can you please give @rihr, @bertearazvan, @elbys, @rosudhi and me write access to your repo? Then we can create feature branches here in the future.

File: packages/react-components-lab/docs/CHANGELOG.md (hunk 3 above)
Comment (bertearazvan) [reply packages/react-components-lab/docs/CHANGELOG.md:1]: Yes, we have used that repo/package as a working proof of concept. Once it's merged, I'll make sure to fix the references

**Comment by FranzThomsen1089** [comment]
Will do tomorrowSent from my Galaxy
-------- Original message --------From: Jonas ***@***.***> Date: 06/04/2021  17:59  (GMT+10:00) To: DHI/react-components ***@***.***> Cc: Franz Thomsen ***@***.***>, Mention ***@***.***> Subject: Re: [DHI/react-components] Implement two-package structure (#86) 
Thanks for the reviews @goneale and @FranzThomsen1089.
The PR already incorporates the latest commits on master, so you can merge it at your earliest convenience and we will do the remaining (repo URL) clean-up very soon after.
Can you please give @rihr, @bertearazvan, @elbys, @rosudhi and me write access to your repo? Then we can create feature branches here in the future.

**Comment by FranzThomsen1089** [comment]
Can you please ask thb@dhigroup.com to be enrolled in the DHI organisation if you are not already

**Comment by bertearazvan** [comment]
Now that we have the access to merge and the PR is generally approved, it would be a good idea to fix all those DHI-GRAS references before merging. I will also temporarily suspend the pipeline trigger for the lab until we decide under what name to publish the lab package and all these details. I will come with some changes ASAP to fix it

- [ ] Replace the references containing DHI-GRAS/react-components-ts
- [ ] Suspend pipeline for the time being until we make some more decisions onto how to proceed

**Comment by FranzThomsen1089** [comment]
@j08lue I have made you admin as well. As we are approaching the limit of how many individual users we can add, I would suggest we create a team like the DomainServices team but for react dev and then add that. Is that something you could have a look at? Also, if the PR is done, I will be happy to approve it :-)

**Comment by j08lue** [comment]
Thanks @FranzThomsen1089! The PR is ready for merging from our side now.

**Comment by j08lue** [comment]
I merged the latest additions from master again. Looking forward to your review @JeanLoureiro, please let me know if any questions or obstacles come up.

File: .vscode/settings.json [hunk 4]
```diff
@@ -1,9 +1,15 @@
...
-  "editor.codeActionsOnSave": {
-    "source.organizeImports": true
-  },
-  "editor.formatOnSave": true,
-  "eslint.format.enable": true,
-  "typescript.tsdk": "node_modules/typescript/lib",
-  "workbench.tree.indent": 20
+    "editor.codeActionsOnSave": {
+        "source.organizeImports": false,
+        "source.fixAll.eslint": true,
```
Comment (JeanLoureiro) [inline .vscode/settings.json:4]: FYI: @goneale and @FranzThomsen1089 just be aware that we might or might not have more code changes next we save soemthing. 

this is the step in the right direction thou.

File: packages/react-components-lab/src/components/Snackbar/SnackbarProvider.tsx [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+        transition = Slide;
+        break;
+      default:
+        transition = Slide;
+    }
+    return transition;
+  };
+
+  return (
+    <>
```
Comment (JeanLoureiro) [inline packages/react-components-lab/src/components/Snackbar/SnackbarProvider.tsx:140]: I can see Fragment shortcut in a couple places where we don't need them. Is it by choice?

**Comment by JeanLoureiro** [comment]
@j08lue 
Jonas, I am happy with it, just placed a few comments, nothing major. 🙂

**Review by JeanLoureiro — APPROVED** [review]

File: packages/react-components-lab/src/components/Snackbar/SnackbarProvider.tsx (hunk 5 above)
Comment (rosudhi) [reply packages/react-components-lab/src/components/Snackbar/SnackbarProvider.tsx:140]: Ah, you're right. MuiSnackbar was outside SnackbarContext.Provider wrapper, then i did some enhancement by moved it inside but forget to remove fragment wrapper. Will do that in next version.
There is other in stories file which want to add two or more component inside initially.

Comment (j08lue) [reply packages/react-components-lab/src/components/Snackbar/SnackbarProvider.tsx:140]: I opened an issue for this, so we can keep track of fixing this.

**Pull request merged**